src/python/arcor2_build/source/python_to_json.py

```
@@ -0,0 +1,420 @@
+
+                for ap in project.action_points:  # action_point value
+                    if (
+                        isinstance(arg, Attribute)
+                        and isinstance(arg.value, Attribute)
+                        and isinstance(arg.value.value, Attribute)
+                    ):
+                        if ap.name == arg.value.value.attr:
+                            action_point = ap
+                            if arg.value.attr == "poses":
+                                param_value = json.dumps(ap.orientations[0].id)
+                            elif arg.value.attr == "joints":
+                                param_value = json.dumps(ap.robot_joints[0].id)
+                            else:
+                                param_value = json.dumps(ap.id)
+
+        elif isinstance(arg, Name):  # variable definated in script or variable from project parameters
+            arg_name = ast.unparse(arg)
+            if arg_name in variables:
+                param_type = ActionParameter.TypeEnum.LINK
+                param_value = json.dumps(f"{variables[arg_name]}/default/0")
+            else:
+                for param in project.parameters:
+                    if arg.id == param.name:
+                        param_type = ActionParameter.TypeEnum.PROJECT_PARAMETER
+                        param_value = json.dumps(param.id)
+        else:
+            raise Arcor2Exception("Unsupported operation.")
+        parameters += [ActionParameter(param_name, param_type, param_value)]
+
+    return parameters, action_point
+
+
+def get_type(rest_of_node: Call, scene: Scene):
+    object_method = ast.unparse(rest_of_node.func)
+    # looking for type of action
+    for object in scene.objects:
+        if object_method.find(object.name + ".") != -1:
+            # raplece name from code to scene definition
+            type = object_method.replace(object.name + ".", (object.id + "/"))
+
+            return type, object_method
+
```

You should raise an exception if you don't find it...